src/Utilities/TrackedDependencies/FileTracker.cs

```diff
@@ -333,40 +303,26 @@ public static string FormatRootingMarker(ITaskItem[] sources)
         /// <param name="outputs">ITaskItem array of outputs.</param>
         public static string FormatRootingMarker(ITaskItem[] sources, ITaskItem[] outputs)
```

Comment:
I did some actual refactoring in this method to avoid some allocations. Might want to take a closer look at this method.

Reply:
separate commit, please